test/cache.js

```diff
@@ -30,16 +31,32 @@ test('entries are not cached using buffers from default slab', async function (t
   await db.close()
 })
 
-test('maxCacheSize arg can be set', async function (t) {
-  const core = new Hypercore(RAM.reusable())
-  const db = new Hyperbee(core, { maxCacheSize: 10 })
+test('node and key caches are subbed from a passed-in rache', async t => {
+  const globalCache = new Rache()
+  const core = new Hypercore(RAM, { globalCache })
+  console.log('core has', core.globalCache)
+  const db = new Hyperbee(core)
+
+  t.is(globalCache.globalSize, 0, 'sanity check')
+  await db.put('some', 'thing')
+  await db.get('some')
 
-  t.is(db.maxCacheSize, 10, 'Correct max cache size')
+  // TODO: for some reason, there's only 1 cache entry
+  // total after a put+get, which seems off. Investigate.
+  t.is(globalCache.globalSize > 0, true, 'subbed from globalCache')
 })
 
-test('default maxCacheSize', async function (t) {
-  const core = new Hypercore(RAM.reusable())
+test('node and key caches are derived from the same rache instance', async t => {
+  const core = new Hypercore(RAM)
   const db = new Hyperbee(core)
-
-  t.is(db.maxCacheSize, 65536, 'Correct max cache size')
+  await db.ready()
+
+  // Note: not an ideal test, since it accesses private props,
```

Comment:
This test can just be removed before getting this PR in. I left it here for now because it illustrates the behaviour I commented in the PR description, for the case where `core.globalCache` is not set.